internal/dag/cache.go

```diff
@@ -224,6 +225,9 @@ func (kc *KubernetesCache) Insert(obj interface{}) bool {
 	case *contour_api_v1.TLSCertificateDelegation:
 		kc.tlscertificatedelegations[k8s.NamespacedNameOf(obj)] = obj
 		return true
+	case *gatewayapi_v1alpha1.GatewayClass:
```

Comment:
I think we need some cases in cache_test.go for this files changes

Reply:
ack